core/trino-main/src/main/java/io/trino/operator/RowIdHashStrategy.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+/**
+ * Hash strategy that evaluates over row IDs
+ */
```

Comment:
do we need explicit `RowIdHashStrategy` and `RowIdComparisonStrategy`?

Reply:
You can lump RowIdHashStrategy and RowIdComparisonStrategy together, but the APIs themselves are used in distinct places, which makes it easier to test separately. Also giving the context, I think, it provides some clarity and consistency on what to expect here.